app/src/main/java/com/sun/chat_04/ui/signup/SignUpActivity.kt

```diff
@@ -13,32 +13,106 @@ import android.view.View
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.FirebaseDatabase
 import com.jpardogo.android.googleprogressbar.library.ChromeFloatingCirclesDrawable
+import com.sun.chat_04.R
 import com.sun.chat_04.R.color
 import com.sun.chat_04.R.id
 import com.sun.chat_04.R.layout
 import com.sun.chat_04.data.model.User
-import kotlinx.android.synthetic.main.sign_up_screen.button_signUp
-import kotlinx.android.synthetic.main.sign_up_screen.edit_confirm_password
-import kotlinx.android.synthetic.main.sign_up_screen.edit_email
-import kotlinx.android.synthetic.main.sign_up_screen.edit_fullname
-import kotlinx.android.synthetic.main.sign_up_screen.edit_password
+import com.sun.chat_04.data.remote.UserRemoteDataSource
+import com.sun.chat_04.data.repositories.UserRepository
+import kotlinx.android.synthetic.main.sign_up_screen.buttonSignUp
+import kotlinx.android.synthetic.main.sign_up_screen.editConfirmPassword
+import kotlinx.android.synthetic.main.sign_up_screen.editEmail
+import kotlinx.android.synthetic.main.sign_up_screen.editFullname
+import kotlinx.android.synthetic.main.sign_up_screen.editPassword
 import kotlinx.android.synthetic.main.sign_up_screen.progress
-import kotlinx.android.synthetic.main.sign_up_screen.radio_groups
-import kotlinx.android.synthetic.main.sign_up_screen.text_birthday
+import kotlinx.android.synthetic.main.sign_up_screen.radioGroups
+import kotlinx.android.synthetic.main.sign_up_screen.textBirthday
 import kotlinx.android.synthetic.main.sign_up_screen.toolbar
 import java.util.Calendar
 
-class SignUpActivity : AppCompatActivity(), OnClickListener {
-    private var user: User? = null
+class SignUpActivity : AppCompatActivity(), OnClickListener, SignUpContract.View {
+
+    private var presenter: SignUpPresenter? = null
+    private var user: User = User()
+    private var email = ""
+    private var password = ""
+    private var confirmPasswork = ""
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(layout.sign_up_screen)
+        val firebaseAuth = FirebaseAuth.getInstance()
+        val firebaseDatabase = FirebaseDatabase.getInstance()
+        presenter = SignUpPresenter(this, UserRepository(UserRemoteDataSource(firebaseAuth, firebaseDatabase)))
         initComponents()
         requirePermissions()
     }
 
+    override fun isSaved(status: Boolean) {
+        if (status) {
+            showNotify(resources.getString(R.string.sign_up_success))
+        } else {
+            showNotify(resources.getString(R.string.sign_up_existed))
+        }
+        progress.visibility = View.GONE
+    }
+
+    override fun onClick(v: View?) {
+        when (v?.id) {
+            id.textBirthday -> showDatePicker()
+            id.buttonSignUp -> handleSignUp()
+        }
+    }
+
+    override fun validateUser() {
+        presenter?.signUp(user, email, password, confirmPasswork)
+    }
+
+    override fun isEmptyUserName() {
```

Comment:
```suggestion
    override fun onEmptyUserName() {
```
các hàm is Abx...
thường trả về boolean